site/blog/2016/jul/1/sphinx-and-rtd-for-writers.rst

```diff
@@ -0,0 +1,552 @@
+
+    .. post:: 2016-03-15 09:00
+       :tags: writing, stc, sphinx
+
+This adds it to the set of posts that get shown in any document where you use::
+
+    .. postlist:: 
+       :tags: stc
+
+You can even create specific archive pages for categories or tags with ``postlist``,
+or just output all of your posts.
+
+This shows some of the magical things you can do with Sphinx's extensibility.
+If you're curious,
+this article was actually written in reStructuredText and then exported to Word for publishing.
+You can see the RST in full here: https://github.com/ericholscher/ericholscher.com/blob/master/site/blog/2016/jul/1/sphinx-and-rtd-for-writers.rst
+
+Custom builders
+---------------
+
+Sphinx supports custom builders to perform tasks beyond providing basic output formats.
+Some examples that ship with Sphinx:
+
+* A linkcheck builder that tells you about broken URL's
+* A builder that outputs all the things changed in the latest version of your code for a changelog
+* An XML builder that outputs a representation of your documents in XML
+* A Man page builder that builds man pages from your documentation
+* JSON builder that outputs your pages as HTML inside of JSON, with some metadata, for embedding dynamically
+
+You can get as creative as you like with custom builders,
+as many current members of the Sphinx community have already done.
+
+Tradeoffs with Sphinx
+---------------------
+
+Every tool has its issues and limitations.
+I'd like to address some of the issues with Sphinx,
+so that you can go into it with eyes wide open.
+
+The biggest issue is that it is originally a *programmer tool*.
+This means that a lot of the designs assume knowledge of code,
+especially for installation and extending the tools.
+Knowledge of the command line is definitely required.
+
+The markup language,
+reStructuredText,
+is also a bit finicky.
+It depends on whitespace for separation of content,
+which is a natural concept for Python programmers,
+but not for most writers.
+
+In general though,
+a lot of the complexity in the language comes from the extensibility and power.
+When compared to something like Markdown,
+reStructuredText can do so much more that it's worth the complexity and somewhat steep learning curve.
+
+Introduction to Read the Docs
+-----------------------------
+
+Read the Docs is a hosting platform for Sphinx-generated documentation.
+It takes the power of Sphinx and adds version control, 
+full-text search,
+and other useful features.
+It pulls down code and doc files from Git,
+Mercurial,
+or Subversion, 
+then builds and hosts your documentation.
+We'll use GitHub in this example as it's the most commonly used system for accessing code.
+
+To get started,
+you create a Read the Docs account,
+and link your GitHub account.
+Then you select the GitHub repository you'd like to build documentation for,
+at which point the magic happens.
+
+Read the Docs:
```

Comment:
`Read the Docs:` -> `... at which point the magic happens. Read the Docs will:`

`Read the Docs:` implies a feature list. not a description of the process.